src/lib/provable/crypto/foreign-twisted.ts

```diff
@@ -0,0 +1,309 @@
+import {
+  TwistedCurveParams,
+  createCurveTwisted,
+} from '../../../bindings/crypto/elliptic-curve.js';
+import type { Group } from '../group.js';
+import { ProvablePureExtended } from '../types/struct.js';
+import { AlmostForeignField, createForeignField } from '../foreign-field.js';
+import { CurveTwisted, Point } from '../gadgets/twisted-curve.js';
+import { Field3 } from '../gadgets/foreign-field.js';
+import { assert } from '../gadgets/common.js';
+import { Provable } from '../provable.js';
+import { provableFromClass } from '../types/provable-derivers.js';
+import { l2Mask, multiRangeCheck } from '../gadgets/range-check.js';
+import { Bytes } from '../bytes.js';
+
+// external API
+export { createForeignTwisted, ForeignTwisted };
+
+// internal API
+export { toPoint, FlexiblePoint };
+
+type FlexiblePoint = {
+  x: AlmostForeignField | Field3 | bigint | number;
+  y: AlmostForeignField | Field3 | bigint | number;
+};
+function toPoint({ x, y }: ForeignTwisted): Point {
+  return { x: x.value, y: y.value };
+}
+
+class ForeignTwisted {
```

Comment:
Is this code supposed to be generic for any Edwards curve given in affine coordinates?

Reply:
The code in the bindings can be used for any curve, but in the provable side of things, we are currently only supporting the edwards25519 curve (but can be easily extended in the future). Thus, I can write a disclaimer on top explaining this.